### PR TITLE
feat(autoware_adapi_v1_msgs): add operation mode api message

### PR DESCRIPTION
**Pull request by isamu-takagi**
## Description

Add operation mode message.
https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/ad-api/list/api/operation_mode/

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_adapi_v1_msgs/operation_mode/msg/OperationModeState.msg [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# constants for mode
+uint16 UNKNOWN = 0
+uint16 STOP = 1
+uint16 AUTONOMOUS = 2
+uint16 LOCAL = 3
+uint16 REMOTE = 4
+
+# variables
+builtin_interfaces/Time stamp
+uint16 mode
```
Comment (kenji-miyake) [inline autoware_adapi_v1_msgs/operation_mode/msg/OperationModeState.msg:10]: Are there any reasons for using `uint16` instead of `uint8` here?

**Review by kenji-miyake — APPROVED** [review]

File: autoware_adapi_v1_msgs/operation_mode/msg/OperationModeState.msg (hunk 1 above)
Comment (isamu-takagi) [reply autoware_adapi_v1_msgs/operation_mode/msg/OperationModeState.msg:10]: No, there is no particular reason. I used uint16 for the state, so I made it the same.

Comment (kenji-miyake) [reply autoware_adapi_v1_msgs/operation_mode/msg/OperationModeState.msg:10]: I see. In that case, I feel it's better to make it minimum, but I'll leave it to you.

Comment (isamu-takagi) [reply autoware_adapi_v1_msgs/operation_mode/msg/OperationModeState.msg]: Done https://github.com/autowarefoundation/autoware_adapi_msgs/pull/17/commits/0ad9f74d323d0df7b0144f55ab042d9360088f0e.

**Pull request merged**